File: src/backend/routes/indices.py
from fastapi import APIRouter
from services import indices as ind

router = APIRouter(
  prefix="/indices",
  tags=["indices"],
  responses={404: {"description": "Not found"}},
)

@router.get("/list")
def read_indices():
  indices = [item['name'] for item in ind.indices]
  return indices

@router.get("/export/{indice}")
def export_ndvi(indice, image_path, output_path, rotation=0, cut_points=[None, None]):
  return ind.get_indice(indice, image_path, output_path, rotation, cut_points)

File: src/services/indices.js
const BACKEND_URL = "http://localhost:8008/indices"

export async function getList() {
  const res = await fetch(`${BACKEND_URL}/list`)
  const json = await res.json()

  return json
}

export async function getIndices({indice, imagePath, outputPath, rotation, cutPoints}) {
  const res = await fetch(`${BACKEND_URL}/export/${indice}/?image_path="${imagePath}"&output_path=${outputPath}"&rotation=${rotation}&cut_points=${cutPoints}`)
  const json = await res.json()
  if (json.error) {
    throw new Error(json.error)
  }
  return json
}